src/Algorithm.php

```diff
@@ -4,6 +4,9 @@
 
 abstract class Algorithm
 {
+    /** @var string */
+    public $type;
```

Comment:
hmm what is `type` representing here? I can't see any references to "type" in the latest draft https://datatracker.ietf.org/doc/draft-cavage-http-signatures/?include_text=1

It looks like you're using this to switch behaviour - perhaps you might want to think about how to abstract this behaviour in an OO way?

Reply:
Well that was trivially corrected. Perhaps I was over-imagining the potential flows.

Resolved in 6c7f8ed81d4a1e279488d54f2a6ba262b71c1ca7